### ISSUE TITLE
Incorporate tagging permissions from upstream efs-csi-driver addon

### ISSUE THREAD

**Issue by blancharda** [defenseunicorns/delivery-aws-iac#224]
### Who is requesting it?
@blancharda 

### What is being requested?
Incorporate tagging permissions from upstream efs-csi-driver addon

### Why is it needed?
AWS will require additional IAM permissions in order to create EFS resources after June 30th.

### What is the impact? What will happen if the request is not fulfilled?
The `enable_efs` module flag will cease to function if this is not addressed by the target date.

### How do we know that we are done?
The lines linked below will be included in the tagged version of the upstream module referenced in our code -- i.e. the efs-csi-driver IAM policy will include permission to tag resources.

### Description
Per a recent update from AWS, EFS resource creation will require IAM permission to `TagResource`s effective June 30:
```
Action required
We are updating Amazon EFS such that a principal applying tags to an EFS resource on creation (For example, a file system or an access point) must have IAM permissions to apply tags to the resources using the TagResource operation. If a principal attempts to create an EFS resource with tags and does not have permission to add tags, the resource creation will not succeed. For this account, this change will go into effect on June 30, 2023.

We have identified the following list of IAM policies that need to be updated to provide sufficient IAM tagging permissions. If they are not updated, then the API calls will not succeed and you will receive an "Access denied" error when making these API calls starting June 30, 2023.

The following is a list of your affected IAM Policy ARN(s):
arn:aws:iam::<acct>:policy/AmazonEKS_EFS_CSI_Driver_Policy
arn:aws:iam::<acct>:policy/efs-driver
```
> NOTE that the above policies were created by enabling the optional `enable_efs` variable

It appears that [the upstream add-on module](https://github.com/aws-ia/terraform-aws-eks-blueprints/blame/main/modules/kubernetes-addons/aws-efs-csi-driver/data.tf#L66-L80) for the efs-csi-driver has already made the necessary addition. We should make sure to make sure to incorporate the change into our modules before the deadline.

**Comment by blancharda** [comment]
We're currently on [tagged release v4.30.0](https://github.com/defenseunicorns/delivery-aws-iac/blob/main/modules/eks/eks-addons.tf#L10) -- I don't believe the upstream change has been added to a release yet (it's only 2 days old)

**Comment by RothAndrew** [comment]
Hello! Thanks for submitting the issue. Would you mind updating it to add context?

From [CONTRIBUTING.md](https://github.com/defenseunicorns/delivery-aws-iac/blob/main/CONTRIBUTING.md#definition-of-ready-for-a-backlog-item)


> To meet the Definition of Ready the issue needs to answer the following questions:
> - Who is requesting it?
> - What is being requested?
> - Why is it needed?
> - What is the impact? What will happen if the request is not fulfilled?
> - How do we know that we are done?
> 
> This can take various forms, and we don't care which form the issue takes as long as it answers the questions above.

**Comment by blancharda** [comment]
Updated.

**Comment by RothAndrew** [comment]
Thanks!

> The lines linked below will be included in the tagged version of the upstream module referenced in our code -- i.e. the efs-csi-driver IAM policy will include permission to tag resources.

> It appears that [the upstream add-on module](https://github.com/aws-ia/terraform-aws-eks-blueprints/blame/main/modules/kubernetes-addons/aws-efs-csi-driver/data.tf#L66-L80) for the efs-csi-driver has already made the necessary addition. We should make sure to make sure to incorporate the change into our modules before the deadline.

I'm not sure I'm following. Are you saying that this is fixed in the upstream module, and we just need to make sure we are using the latest version? or are there changes we need to make?

**Comment by RothAndrew** [comment]
Gotcha. Renovate will submit a PR automatically as soon as it comes out, we'll make sure to get it merged.

**Comment by RothAndrew** [comment]
The update was released in v4.31.0 which this repo now uses (#221). Closing.
